modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js

```diff
@@ -71,12 +88,37 @@ function Item(props) {
 
 function MiniCompare(props) {
 	const [items, setItems] = useState(props.items);
-
-	compareCookie.setValue(
-		props.commerceChannelGroupId,
-		items.map((item) => item.id).join(':')
+	const [functionalCookiesConsent, setFunctionalCookiesConsent] = useState(
+		false
 	);
 
+	useEffect(() => {
+		setFunctionalCookiesConsent(checkConsent(COOKIE_TYPES.FUNCTIONAL));
+	});
+
+	const CheckProductComparisonCookieAcceptance = () => {
+		const modalOptions = {
+			alertMessage: Liferay.Language.get(
+				'the-compare-function-requires-acceptance-of-functional-cookies'
+			),
+			customTitle: Liferay.Language.get(
+				'product-comparison-uses-non-essential-cookies'
+			),
+		};
+		checkCookieConsentForTypes(COOKIE_TYPES.FUNCTIONAL, modalOptions)
+			.then(() => {
+				compareCookie.setValue(
+					props.commerceChannelGroupId,
+					items.map((item) => item.id).join(':')
+				);
+				setFunctionalCookiesConsent(true);
+			})
+			.catch(() => {
+				alertCookies();
+			});
+		return null;
```

Comment:
As this funcion component delegates rendering to `checkCookieConsentForTypes()`, there's no output to return. This pattern looks a bit hacky to me, I could not find a better way to integrate the API into the react cycle. @kresimir-coko @bryceosterhaus maybe there's a better way?

Reply:
Just so I fully understand, you want to make sure that `checkCookieConsentForTypes` is confirmed before rending the main content of this component?

Reply:
If they deny the cookie, does `setFunctionalCookiesConsent(true);` still get set?

Reply:
yes, we need functional cookies to be accepted before rendering the main content. I'm taking this as a kind of "loading..." state while fetched data comes, with the peculiarity that in some cases there's no need to run any real fetch as the data is already there.

> If they deny the cookie, does setFunctionalCookiesConsent(true); still get set?

It should not as  `setFunctionalCookiesConsent(true)` is in the resolve() part. However, there are a remote chance of not syncing the component state with the cookie status as we do in the initial `useEffect()` call. Maybe we can use:

`setFunctionalCookiesConsent(checkConsent(COOKIE_TYPES.FUNCTIONAL))`

instead

Reply:
I'm starting to think that some sort of `useCookieConsent()` hook might make sense.

Nevertheless, as a side note, we can do that after sending this PR as there's a tight deadline for the cookie management feature (which depends on this product comparison changes) to be released